Sources/Workspace/ManagedDependency.swift

```diff
@@ -79,8 +77,8 @@ public final class ManagedDependency: JSONMappable, JSONSerializable {
 
     /// Create an editable managed dependency based on a dependency which
     /// was *not* in edit state.
-    func makingEditable(subpath: RelativePath, state: State) -> ManagedDependency {
-        return ManagedDependency(basedOn: self, subpath: subpath, state: state)
+    func makingEditable(subpath: RelativePath, path: AbsolutePath?) -> ManagedDependency {
```

Comment:
We should add doc comments for subpath and path here, and maybe label path as umanagedPath (or similar) instead.